### PR TITLE
feat: Override for project configs

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

**Comment by untitaker** [comment]
The apparent motivation for the "passive config mode" feature is to avoid writing a project config UI for sentry, the PR offers an alternative. However, relays still unconditionally authenticate.

File: common/src/config.rs [hunk 1]
```diff
@@ -291,6 +293,40 @@ impl Default for Sentry {
...
+/// Project state that was hardcoded by the user in the local config
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StaticProjectState {
+    /// Indicates that the project is disabled.
+    pub disabled: bool,
+    /// A container of known public keys in the project.
+    pub public_keys: HashMap<String, bool>,
+    /// The project's slug if configured.
+    pub slug: String,
```
Comment (mitsuhiko) [inline common/src/config.rs]: We should make this optional

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Comment by untitaker** [comment]
One ugly thing I found is that `ProjectConfig` and `StaticProjectState` have camelCase keys, while the other keys don't. It seems easiest to change everything to camelcase...

**Comment by untitaker** [comment]
- [ ] make one file per project config

**Pull request closed without merging**